src/main/io/dataEdition.c

```diff
@@ -0,0 +1,1359 @@
+			masterConfig.escAndServoConfig.servoCenterPulse = (structureOfPage.editField[12].data * 10);
+			masterConfig.escAndServoConfig.maxthrottle = (structureOfPage.editField[15].data * 10);
+			masterConfig.escAndServoConfig.mincommand = (structureOfPage.editField[18].data * 10);
+		    break;
+		}
+	    case PIDCONTROLLER_PAGE : {
+	    	index = getCurrentProfile();
+	    	index = firstItemOnLine(masterConfig.profile[index].pidProfile.pidController + 1);
+	    	structureOfPage.editField[index].data = 0;
+	    	masterConfig.profile[getCurrentProfile()].pidProfile.pidController = selectedLine - 1;
+		    break;
+	    }
+	    case PID_PROFILE_PAGE : {
+			if (selectedItem == 3) {
+				setCurrentPIDprofileConfig(structureOfPage.editField[3].data);
+				updatePIDValuesForThatProfile();
+			}
+			else {
+				index = structureOfPage.editField[3].data;
+				masterConfig.profile[index].pidProfile.P8[ROLL] = structureOfPage.editField[9].data;
+				masterConfig.profile[index].pidProfile.I8[ROLL] = structureOfPage.editField[10].data;
+				masterConfig.profile[index].pidProfile.D8[ROLL] = structureOfPage.editField[11].data;
+				masterConfig.profile[index].pidProfile.P8[PITCH] = structureOfPage.editField[12].data;
+				masterConfig.profile[index].pidProfile.I8[PITCH] = structureOfPage.editField[13].data;
+				masterConfig.profile[index].pidProfile.D8[PITCH] = structureOfPage.editField[14].data;
+				masterConfig.profile[index].pidProfile.P8[YAW] = structureOfPage.editField[15].data;
+				masterConfig.profile[index].pidProfile.I8[YAW] = structureOfPage.editField[16].data;
+				masterConfig.profile[index].pidProfile.D8[YAW] = structureOfPage.editField[17].data;
+			}
+		    break;
+	    }
+	    case RATE_PROFILE_PAGE : {
+			if (selectedItem == 3) {
+				setControlRateProfile(structureOfPage.editField[3].data);
+				updateRateValuesForThatProfile();
+			}
+			else {
+				index = structureOfPage.editField[3].data;
+				masterConfig.controlRateProfiles[index].rates[FD_ROLL] = structureOfPage.editField[9].data;
+				masterConfig.controlRateProfiles[index].rates[FD_PITCH] = structureOfPage.editField[10].data;
+				masterConfig.controlRateProfiles[index].rates[FD_YAW] = structureOfPage.editField[11].data;
+				masterConfig.controlRateProfiles[index].dynThrPID = structureOfPage.editField[15].data;
+				masterConfig.controlRateProfiles[index].tpa_breakpoint = structureOfPage.editField[18].data * 10;
+			}
+		    break;
+	    }
+	    case GPS_PAGE : {
+#ifdef GPS
+	    	testAndPositionThatBit(structureOfPage.editField[3].data, FEATURE_GPS);
+	    	masterConfig.gpsConfig.provider = structureOfPage.editField[6].data;
+	    	gpsData.baudrateIndex = structureOfPage.editField[9].data;
+	    	masterConfig.gpsConfig.sbasMode = structureOfPage.editField[12].data;
+
+            index = getCurrentProfile();
+	    	masterConfig.profile[index].mag_declination = structureOfPage.editField[16].data * 100
```

Comment:
you should probably use local variable to compute new value and write profile only once